server/src/test/java/org/opensearch/cluster/structure/RoutingIteratorTests.java

```diff
@@ -768,4 +770,94 @@ public void testWeightedRoutingShardState() {
             terminate(threadPool);
         }
     }
+
+    /**
+     * Test to validate that shard routing state is maintained across requests, requests are assigned to nodes
+     * according to assigned routing weights
+     */
+    public void testWeightedRoutingShardStateWithDifferentWeights() {
```

Comment:
Wondering how did we miss this regression since we had ITs covering same . If not , lets please add

Reply:
Yeah we have added a test now